### PR TITLE
python312Packages.pip-chill: init at 1.0.3

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Sigmanificient — CHANGES_REQUESTED** [review]
Please rename your commit (and pr title) to `python312Packages.pip-chill: init at 1.0.3`

**Comment by Tochiaha** [comment]
#324722 ssort

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ lib, fetchFromGitHub, buildPythonPackage, bandit, coverage, flake8, nose
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pip-chill/default.nix:1]: Please apply the newer `nixfmt` formatter to this file

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    coverage
+    flake8
+    nose
+    pytest
+    sphinx
+    #ssort
+    tox
+    twine
+    watchdog
+  ] ++ [ bump2version ];
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pip-chill/default.nix:33]: Many of the packages you set as dependencies are tools dev-dependencies, and are not related to the package itself. `pytest` is also provided in check phase by the `pytestCheckHook`, so it shouldnt be listed too.

```suggestion
  dependencies = [ pip ];
```

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pip-chill/default.nix:36]: the tests try to run `pip-chill` as an executable, which should be patched to use the correct path:

```suggestion
  preCheck = ''
    substituteInPlace tests/test_pip_chill.py \
      --replace-fail "pip_chill/cli.py" "${placeholder "out"}/bin/pip-chill"
  '';
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  pythonImportsCheck = [ "pip_chill" ];
+
+  meta = {
+    description = "A more relaxed `pip freeze`";
+    homepage = ";https://github.com/rbanffy/pip-chill";
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ tochiaha ];
+    mainProgram = "pip-chill";
+    platforms = lib.platforms.all;
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/pip-chill/default.nix:45]: I think `platforms` is uneeded

**Comment by Tochiaha** [comment]
@superherointj

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  pythonImportsCheck = [ "pip_chill" ];
+
+  preCheck = ''
+    substituteInPlace tests/test_pip_chill.py \
+      --replace-fail "pip_chill/cli.py" "${placeholder "out"}/bin/pip-chill"
+  '';
+
+  meta = {
+    description = "A more relaxed `pip freeze`";
+    homepage = ";https://github.com/rbanffy/pip-chill";
```
Comment (superherointj) [inline pkgs/development/python-modules/pip-chill/default.nix:31]:
```suggestion
    homepage = "https://github.com/rbanffy/pip-chill";
```

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{ lib, fetchFromGitHub, buildPythonPackage, bump2version, setuptools, pip
```
Comment (superherointj) [inline pkgs/development/python-modules/pip-chill/default.nix:1]: Please use `nixfmt` (rfc).

**Comment by superherointj** [comment]
Please, always squash commits.

**Comment by superherointj** [comment]
According to: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

Add the release note to the body of commit message.

> Release: https://github.com/rbanffy/pip-chill/releases/tag/v1.0.3

**Comment by superherointj** [comment]
Please **only** add to the body of commit message:

> Release: https://github.com/rbanffy/pip-chill/releases/tag/v1.0.3

Yes, the URL, not actual contents of upstream changelog.
You could mention other things you did as well, but not from the upstream POV, but from Nix package POV, things done to the package.

Please understand, this is a simple detail, and it is annoying for you to receive such a tiny feedback, and for me to give you such a tiny feedback, and you might wonder why I'm insisting on such detail, it is to establish a pattern to be followed in all of your PRs. It's in the [documentation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions). This facilitates review, since reviewers and users (you also) can just click on link and go to upstream and check things as necessary.

Thanks for you understanding.

**Comment by Tochiaha** [comment]
@superherointj thank you, for the review, i believe i got the memo now.

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    pip
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "pip_chill" ];
+
+  preCheck = ''
+    substituteInPlace tests/test_pip_chill.py \
+      --replace-fail "pip_chill/cli.py" "${placeholder "out"}/bin/pip-chill"
+  '';
```
Comment (superherointj) [inline pkgs/development/python-modules/pip-chill/default.nix:37]:
```
 preCheck = ''
    export PATH=$out/bin:$PATH
  '';
```
Would this work?

File: pkgs/development/python-modules/pip-chill/default.nix [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  build-system = [ setuptools ];
+
+  dependencies = lib.optionals (pythonAtLeast "3.12") [ setuptools ];
+
+  nativeCheckInputs = [
+    pip
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "pip_chill" ];
```
Comment (superherointj) [inline pkgs/development/python-modules/pip-chill/default.nix:32]: The order of parameter mostly follow the [phases order](https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases). Which is the order of execution.

I would add `pythonImportsCheck` to `installCheck` phase. Meaning, just before meta, in this case.

File: pkgs/development/python-modules/pip-chill/default.nix (hunk 7 above)
Comment (Tochiaha) [reply pkgs/development/python-modules/pip-chill/default.nix:37]: Yes it works

Comment (superherointj) [reply pkgs/development/python-modules/pip-chill/default.nix:37]: Seems a better version then, smaller and uncoupled from specific files.

Comment (Sigmanificient) [reply pkgs/development/python-modules/pip-chill/default.nix:37]: Yeah, I didn't thought of this

Comment (superherointj) [reply pkgs/development/python-modules/pip-chill/default.nix:37]: > Yeah, I didn't thought of this

Neither did I. The nixpkgs collective is as good as the [Borg](https://en.wikipedia.org/wiki/Borg). Resistance is futile.

**Comment by superherointj** [comment]
Please squash commits.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325625` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.pip-chill</li>
    <li>python311Packages.pip-chill.dist</li>
    <li>python312Packages.pip-chill</li>
    <li>python312Packages.pip-chill.dist</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325625` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.pip-chill</li>
    <li>python311Packages.pip-chill.dist</li>
    <li>python312Packages.pip-chill</li>
    <li>python312Packages.pip-chill.dist</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Can you check if it is building?
Also, consider rebasing to master, just in case.
- python312Packages.pip-chill => https://termbin.com/9b9r

**Comment by Tochiaha** [comment]
> Can you check if it is building? Also, consider rebasing to master, just in case.
> 
>     * python312Packages.pip-chill => https://termbin.com/9b9r

it builds but produces failed_result directory with the output in the directory.

**Review by superherointj — CHANGES_REQUESTED** [review]
Package build should be successful.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325625` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.pip-chill</li>
    <li>python311Packages.pip-chill.dist</li>
    <li>python312Packages.pip-chill</li>
    <li>python312Packages.pip-chill.dist</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325625` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.pip-chill</li>
    <li>python311Packages.pip-chill.dist</li>
    <li>python312Packages.pip-chill</li>
    <li>python312Packages.pip-chill.dist</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325625` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.pip-chill</li>
    <li>python311Packages.pip-chill.dist</li>
    <li>python312Packages.pip-chill</li>
    <li>python312Packages.pip-chill.dist</li>
  </ul>
</details>

**Comment by Sigmanificient** [comment]
hmm what do we do about this pr?

**Pull request closed without merging**